Automattic/vip-go-mu-plugins
# Fix OOM errors on large video uploads

## Description

Uploading large video files(size close to 1GB), results in an Out of memory Fatal error  in the post-upload meta processing stage. Specifically, in the call to `wp_read_video_metadata()`.

Root cause appears to be the use of `file_get_contents()` within the stream wrapper cache class. This PR changes how the local cache works by returning a stream instead of the file content to resolve the OOM error.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Check out PR.
1. Go to `wp-admin` > `Media` > `Library`
1. Upload a video file that is close to the 1GB limit
1. Verify no errors occured.
